lib/test_cleanup.rb

```diff
@@ -104,11 +106,19 @@ def cleanup_shared
 
     private
 
-    def checkout_branch_if_needed(repository, branch = "master")
+    def default_origin_ref(repository)
+      default_branch = Utils.popen_read(
+        git, "-C", repository, "symbolic-ref", "refs/remotes/origin/HEAD", "--short"
+      ).presence ||= "origin/master"
+      default_branch.strip
```

Comment:
```suggestion
      default_branch = Utils.popen_read(
        git, "-C", repository, "symbolic-ref", "refs/remotes/origin/HEAD", "--short"
      ).strip.presence
      default_branch ||= "origin/master"
      default_branch
```

Reply:
Won't strip fail if the command fails? (My ruby knowledge is failing me)

Reply:
No, it'll always return a string.